src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
+        r = (pos - origin).norm
+        c = (r < radius).values
+        if np.any(c):
+            subdomain[name] = dataset[name][c]
+
+    return subdomain
+
+
+def extract_box(dataset, dx, dy, dz, origin):
```

Comment:
What will be the most common you think? supplying a center and a dx, or supplying xmin, xmax, ymin, ymax etc...

Reply:
Now that I think of it, I feel that `origin` is a bit ambiguous here. When we say `origin = 0` and `dx = 100`, do we mean between `-50 < x < 50` or do we mean `0 < x < 100` ?
Maybe we should only support `xmin, xmax, ymin, ymax, zmin, zmax`?

Alternatively, we could rename `origin` to `center`? Maybe that would make it less ambiguous?

Reply:
I'd rather have `dx, dy, dz`, it's a more compact way of doing things.

> Now that I think of it, I feel that origin is a bit ambiguous here. When we say origin = 0 and dx = 100, do we mean between -50 < x < 50 or do we mean 0 < x < 100 ?

We mean the first one, we can explain this in the documentation which I'll push with #106 (or alternatively, a separate PR).

> Alternatively, we could rename origin to center? Maybe that would make it less ambiguous?

I'm not so sure about this. `origin` here is the mathematical meaning of the word (see [here](https://en.wikipedia.org/wiki/Origin_(mathematics))), I feel it would be more ambiguous to have `center`.

Reply:
> origin here is the mathematical meaning of the word

That might be, but `dx` does not necessarily mean we start at `-dx/2`. But it's probably ok for now, because we use the same terminology in `map`.